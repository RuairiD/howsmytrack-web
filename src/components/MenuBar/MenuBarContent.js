import React, { useState } from "react";

import { A, Div, Img } from "lemon-reset";
import MobileMenu from "./MobileMenu";
import MainMenu from "./MainMenu";

const MenuBarContent = ({
    isMobile,
    onMenuClick,
    username,
    rating,
    notifications,
}) => {
    const [mobileMenuCollapsed, setMobileMenuCollapsed] = useState(true);

    const onCollapseChange = (key) => {
        setMobileMenuCollapsed(key.length === 0);
    };

    return (
        <Div className="menu-bar">
            {!isMobile && (
                <A href="/" className="full-logo">
                    <Img
                        alt="how's my track"
                        src="/logo512.png"
                        width="200px"
                        style={{
                            padding: "0.5em",
                            marginLeft: "auto",
                            marginRight: "auto",
                        }}
                    />
                </A>
            )}
            {isMobile && (
                <MobileMenu
                    onMenuClick={onMenuClick}
                    username={username}
                    rating={rating}
                    notifications={notifications}
                    onCollapseChange={onCollapseChange}
                    mobileMenuCollapsed={mobileMenuCollapsed}
                />
            )}
            {!isMobile && (
                <MainMenu
                    onMenuClick={onMenuClick}
                    username={username}
                    rating={rating}
                    notifications={notifications}
                />
            )}
        </Div>
    );
};

export default MenuBarContent;
