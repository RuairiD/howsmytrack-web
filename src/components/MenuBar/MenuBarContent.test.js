import React from "react";
import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { act } from "react-dom/test-utils";
import MenuBarContent from "./MenuBarContent";

describe("MenuBarContent", () => {
    it("renders logo and MainMenu for non-mobile clients", () => {
        const wrapper = shallow(
            <MenuBarContent />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders a MobileMenu for mobile clients", () => {
        const wrapper = shallow(
            <MenuBarContent isMobile />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("keeps track of menu collapsed state on mobile clients", () => {
        const wrapper = mount(
            <MenuBarContent isMobile />,
        );

        act(() => wrapper.find("MobileMenu").get(0).props.onCollapseChange([]));
        wrapper.update();
        expect(wrapper.find("MobileMenu").get(0).props.mobileMenuCollapsed).toBeTruthy();

        act(() => wrapper.find("MobileMenu").get(0).props.onCollapseChange([1]));
        wrapper.update();
        expect(wrapper.find("MobileMenu").get(0).props.mobileMenuCollapsed).not.toBeTruthy();
    });
});
