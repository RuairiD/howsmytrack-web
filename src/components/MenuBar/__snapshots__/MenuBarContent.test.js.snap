// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuBarContent renders a MobileMenu for mobile clients 1`] = `
<Div
  className="menu-bar"
>
  <MobileMenu
    mobileMenuCollapsed={true}
    onCollapseChange={[Function]}
  />
</Div>
`;

exports[`MenuBarContent renders logo and MainMenu for non-mobile clients 1`] = `
<Div
  className="menu-bar"
>
  <A
    className="full-logo"
    href="/"
  >
    <Img
      alt="how's my track"
      className=""
      height="200px"
      src="/logo512.png"
      style={
        Object {
          "marginLeft": "auto",
          "marginRight": "auto",
          "padding": "0.5em",
        }
      }
      width="200px"
    />
  </A>
  <MainMenu />
</Div>
`;
